test/FluentMigrator.Tests/Unit/Generators/SQLite/SQLiteDataTests.cs

```diff
@@ -28,21 +29,16 @@ namespace FluentMigrator.Tests.Unit.Generators.SQLite
     // ReSharper disable once InconsistentNaming
     public class SQLiteDataTests : BaseDataTests
     {
-        protected SQLiteGenerator Generator;
-
-        [SetUp]
-        public void Setup()
-        {
-            Generator = new SQLiteGenerator();
-        }
+        private static SQLiteGenerator CreateFixture(QuoterOptions options = null) =>
```

Comment:
I wonder if BaseDataTests should be generic and have a `BaseDataTests<T> where T : IGenerator` so that CreateFixture is protected static:

```c#
protected static T CreateFixture(QuoterOptions options = null) =>
```

@fubar-coder It sounded like you didn't like the CreateFixture() thing, vs. using NUnit's well-defined SetUp - but I kind of like it because the style is more in-line with XUnit which I am used to.  I don't really love a distinct SetUp phase.  The one use case that SetUp provides is when working with cancellable Tasks, and needing to memoize the cancellationtoken for the duration of the test.  Just my opinion.

Reply:
I prefer xUnit as well. I don't think that having `CreateFixture` in `BaseDataTests` will result in a maintainable method because CreateFixture might accept other parameters - not only `QuoterOptions`.
Usually, for each class a create an abstract test class where I define the `CreateFixture` method. It this case it would mean adding another abstract class between SQLiteDataTests and BaseDataTests which is an overkill (isn't it?).
Well, therefore, I think having `CreateFixture` is the same class is OK. But please let me know the final decision. I will do it as it suits the FluentMigrator :)